lighthouse-extension/app/src/lighthouse-background.js

```diff
@@ -73,6 +74,33 @@ window.runLighthouse = function(options, requestedAudits) {
 };
 
 /**
+ * @param {!RawProtocol.Port} port
+ * @param {string} url
+ * @param {!Object} options Lighthouse options.
+ * @param {!Array<string>} requestedAudits Names of audits to run.
+ * @return {!Promise}
+ */
+window.runLighthouseInWorker = function(port, url, options, requestedAudits) {
+  // Default to 'info' logging level.
+  log.setLevel('info');
+  const connection = new RawProtocol(port);
+
+  // Always start with a freshly parsed default config.
+  const runConfig = JSON.parse(JSON.stringify(defaultConfig));
+
+  // Filter out audits not requested.
+  requestedAudits = new Set(requestedAudits);
+  runConfig.audits = runConfig.audits.filter(audit => requestedAudits.has(audit));
+  const config = new Config(runConfig);
+
+  // Add url and config to fresh options object.
+  const runOptions = Object.assign({}, options, {url, config});
+
+  // Run Lighthouse.
+  return Runner.run(connection, runOptions);
```

Comment:
can you extract 88-100 into a separate method so we don't duplicate between these two?